justfile

```diff
@@ -178,6 +181,35 @@ cni-plugin-test-integration-calico:
         K3D_CREATE_FLAGS='{{ _K3D_CREATE_FLAGS_NO_CNI }}' \
         cni-plugin-test-integration
 
+cni-plugin-test-integration-cilium:
+    @{{ just_executable() }} \
+        CNI_TEST_SCENARIO='cilium' \
+        K3D_CLUSTER_NAME='l5d-cilium-test' \
+        K3D_CREATE_FLAGS='{{ _K3D_CREATE_FLAGS_NO_CNI_NO_POLICY_ENFORCER }}' \
+        cni-plugin-test-integration-with-cilium
+
+_cni-plugin-setup-cilium:
+    #!/usr/bin/env bash
+    set -euxo pipefail
+    docker exec k3d-l5d-cilium-test-server-0 mount bpffs /sys/fs/bpf -t bpf
+    docker exec k3d-l5d-cilium-test-server-0 mount --make-shared /sys/fs/bpf
+    docker exec k3d-l5d-cilium-test-server-0 mkdir -p /run/cilium/cgroupv2
+    docker exec k3d-l5d-cilium-test-server-0 mount -t cgroup2 none /run/cilium/cgroupv2
+    docker exec k3d-l5d-cilium-test-server-0 mount --make-shared /run/cilium/cgroupv2/
```

Comment:
could probably do all of these in one go:

```
docker exec k3d-l5d-cilium-test-server-0 "mount bpffs /sys/fs/bpf -t bpf; \
 mount --make-shared /sys/fs/bpf;\
 ...
"
```

Reply:
I think these need to be on separate lines in order for the operations to occur in-order. I don't think mount is seeing the result if mkdir and here's the error I get when I combine them:

```+ docker exec k3d-l5d-cilium-test-server-0 'mount bpffs /sys/fs/bpf -t bpf;    mount --make-shared /sys/fs/bpf;    mkdir -p /run/cilium/cgroupv2;    mount -t cgroup2 none /run/cilium/cgroupv2;    mount --make-shared /run/cilium/cgroupv2/'
OCI runtime exec failed: exec failed: unable to start container process: exec: "mount bpffs /sys/fs/bpf -t bpf;    mount --make-shared /sys/fs/bpf;    mkdir -p /run/cilium/cgroupv2;    mount -t cgroup2 none /run/cilium/cgroupv2;    mount --make-shared /run/cilium/cgroupv2/": stat mount bpffs /sys/fs/bpf -t bpf;    mount --make-shared /sys/fs/bpf;    mkdir -p /run/cilium/cgroupv2;    mount -t cgroup2 none /run/cilium/cgroupv2;    mount --make-shared /run/cilium/cgroupv2/: no such file or directory: unknown